### ISSUE TITLE
Firebase background push notification: this.router.navigate doesn't do anything

### ISSUE THREAD

**Issue by vincentriche** [EddyVerbruggen/nativescript-plugin-firebase#1584]
Hi, 
I don't know if it's the right place to post this message but after passing many hours on google to find a solution, I'm quite desperate.

I use firebase push notifications and in the onMessageReceivedCallback function, I navigate to one or another page (with this.router.navigate and ngZone.run() cause i am out the angular zone in here).
When the app is close or in foreground, I don't have any problems the router.navigate is working. But when the app is in background (not stopped, just closed), I click on the new notification, the app open, an nothing happen. Like other cases, i want to navigate to my specific page (when all is loaded).
I don't have any errors but nothing happens, I'm on the main page.

I don't know what to do. I tried to store the context, use a setTimeout, but still the same.

Thanks

**Comment by manojdcoder** [comment]
The plugin has nothing to do with navigation. As long you are running the code once app is foreground and within zone, navigation should work. If you still have issues, please share a sample project.

**Comment by vincentriche** [comment]
Yes i know, but like i try to navigate in a firebase function, maybe someone has an answer..

The onMessageReceived callback is called after the activityResumed event so i suppose the navigate should work but it doesn't. 
I am not in the angular zone cause i need to use NgZone.run() (and it works when the app was claused or in foreground).

**Comment by vincentriche** [comment]
It's hard to share a sample project cause i need to create a firebase, implement firebase push notification, etc. ... :/

**Comment by lostation** [comment]
Are you sure the onMessageReceivedCallback is triggered ? Cause I'm exactly into the same issue...in foreground it works...but not if I tap on push notification from the tray...and app is in background. This onMessageReceivedCallback is never called...

**Comment by vincentriche** [comment]
Yeah I'm sure, thanks to the console log ! But I really don't have any ideas why the navigate just not works in this case ><

**Comment by lostation** [comment]
Where do you init the firebase ? into you app.component.ts or later in a sub component ?

**Comment by vincentriche** [comment]
yes in the ngOnInit of app.component.ts

**Comment by lostation** [comment]
I tried init firebase later from sub component listening for the @HostListener('loaded') event...but it still doesn't work.

**Comment by lostation** [comment]
And what if you try to put your navigation into a setTimeout ?

**Comment by vincentriche** [comment]
Like you i tried, but it still doesn't work ^^'

**Comment by lostation** [comment]
How are you listening the activityResumed ?

**Comment by vincentriche** [comment]
Like this (in the ngOnInit of app.component.js) :
 `application.android.on(application.AndroidApplication.activityResumedEvent, function (args) {
        console.log("Event: " + args.eventName + ", Activity: " + args.activity);
    });
`

**Comment by lostation** [comment]
I really don't understand why at your side, you get the  onMessageReceivedCallback... me not at all... Damn it's really strange.

**Comment by curtiscarlson** [comment]
Yo, I think I solved this by adding a setTimeout for one second around the navigation.  At least, that is in my code and I can't remember why but I think this was it.  I think basically you need to give the app time to load before calling the function.  gotta love js!

**Comment by JavidMohajery** [comment]
I have the same problem. Does someone have solved this?

**Comment by addahlbe** [comment]
I realize I'm late to the party, but Ill answer in case someone else struggles with this in the future. 
I made an observable in my notification service where I add the event listener. Then within the ng zone I emit the value of the notification.
![image](https://user-images.githubusercontent.com/2336335/190176116-4f252884-3be6-44d3-abb3-903458031376.png)

![image](https://user-images.githubusercontent.com/2336335/190175147-a95ed17e-a2dd-4f00-ade8-beb651ad3d7c.png)

Then within my app component, I have a listener to the observable after everything is registered. You can route when it fires here. Mine has some more complicated logic with multi tenant login so I omitted that. But this should help get the point across

![image](https://user-images.githubusercontent.com/2336335/190175707-28fdb5b3-3fee-4a23-aaa8-38a6ddd9fbad.png)

The reason the routing does not work, is you are routing outside the angular zone. This should help put you back inside the angular zone with proper timing (everything loading first).
